src/Type/Constant/ConstantArrayType.php

```diff
@@ -220,6 +220,32 @@ public function getValueTypes(): array
 		return $this->valueTypes;
 	}
 
+	public function getFirstValueType(): Type
+	{
+		$valueTypes = [];
+		for ($i = 0, $keyTypesCount = count($this->keyTypes); $i < $keyTypesCount; $i++) {
+			$valueTypes[] = $this->valueTypes[$i];
+			if (!$this->isOptionalKey($i)) {
```

Comment:
I am not sure whether its obvious that this method skips optionalKeys.

maybe its worth to either document that in phpdoc, adding a bool whether optionals should be skipped or rename the method reflecting it..?

Reply:
https://github.com/phpstan/phpstan-src/pull/1668#discussion_r959269214 :)